diesel_migrations/migrations_macros/src/embed_migrations.rs

```diff
@@ -78,7 +78,13 @@ pub fn derive_embed_migrations(input: &syn::DeriveInput) -> quote::Tokens {
 }
 
 fn migration_literals_from_path(path: &Path) -> Result<Vec<quote::Tokens>, Box<Error>> {
-    try!(migration_paths_in_directory(path))
+    let mut migrations = migration_paths_in_directory(path)?;
+
+    // Sort the migrations to make sure embedded migrations are deterministic
+    // and the binary is reproducible.
+    migrations.sort_by(|a, b| a.path().cmp(&b.path()));
```

Comment:
What do you think about `migrations.sort_by_key(|a| a.path())` instead?

Reply:
Good catch, updated my PR.